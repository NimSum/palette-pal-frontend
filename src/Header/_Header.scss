.Header {
  height: 40px;
  background: #1D1C1C;
  border-bottom: 2px solid #4B5266;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header-left {
  display: flex;
  align-items: center;
  opacity: .9;
  text-decoration: none;
}

.logo-img {
  height: 32px;
  margin: 5px;
}

.logo-text {
  font-family: 'Pacifico', sans-serif;
  color: white;
  font-size: 22px;
  font-weight: 300;
}

.nav-link {
  color: white;
  text-decoration: none;
  font-size: 14px;
  margin: 0 20px 0 35px;
  opacity: .8;
  font-weight: 300;
  height: 100%;

  .active {
    opacity: 1;
    font-weight: 400;
    border-bottom: 2px solid blue;
  }
}