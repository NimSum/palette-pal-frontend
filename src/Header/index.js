import React from 'react';
import { Link, NavLink } from 'react-router-dom';

function Header() {
  return (
    <header className="top-bar">
      <Link exact="true" to="/" className="header-left">
        <img className="logo-img" src={require('../images/logo.png')} alt="Palette pal logo of paint palette" />
        <h1 className="logo-text">Palette Pal</h1>
      </Link>
      <div className="header-right">
        <nav className="header-nav">
          <NavLink exact to="/" className="nav-link">Pick New Palette</NavLink>
          <NavLink exact to="/projects" className="nav-link">My Projects</NavLink>
          <NavLink exact to="/login" className="nav-link">Log In / Sign Up</NavLink>
        </nav>
      </div>
    </header>
  );
}

export default Header;
