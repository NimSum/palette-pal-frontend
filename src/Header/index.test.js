import React from 'react';
import Header from './index';
import { shallow } from 'enzyme';

describe('Header', () => {
  let wrapper;
  let instance;
  const mockLogUserOut = jest.fn();
  const mockLogUserIn = jest.fn();
  const mockSignUserUp = jest.fn();

  let localStorageMock = { getItem: () => false };
  Object.defineProperty(window, 'localStorage', { value: localStorageMock });
  

  beforeEach(() => {
    wrapper = shallow(< Header
      logUserOut={mockLogUserOut}
      logUserIn={mockLogUserIn}
      signUserUp={mockSignUserUp}
    />);
    instance = wrapper.instance();
  })
  
  it('should match component Header snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the expected default state', () => {
    expect(wrapper.state()).toEqual({
      showLogin: false,
      showSignup: false
    })
  })

  it('should call logUserOut when the logout link is clicked', () => {
    localStorageMock = {
      getItem: () => JSON.stringify({ user_token: "hi there" })
    }

    Object.defineProperty(window, 'localStorage', { value: localStorageMock });
    
    wrapper = shallow(< Header
      logUserOut={mockLogUserOut}
      logUserIn={mockLogUserIn}
      signUserUp={mockSignUserUp}
    />);

    wrapper.find('.logout-link').simulate('click');

    expect(mockLogUserOut).toHaveBeenCalled();
  })

  it('should update state when the login link is clicked', () => {
    wrapper.find('.login-link').simulate('click');

    expect(wrapper.state()).toEqual({
      showLogin: true,
      showSignup: false
    })
  })

  it('should update state when the sign up link is clicked', () => {
    wrapper.find('.signup-link').simulate('click');

    expect(wrapper.state()).toEqual({
      showLogin: false,
      showSignup: true
    })
  })
})