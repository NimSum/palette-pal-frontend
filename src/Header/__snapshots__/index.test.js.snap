// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match component Header snapshot 1`] = `ShallowWrapper {}`;
